lib/Command/ExpirePassword.php

```diff
@@ -106,54 +129,110 @@ protected function configure() {
 	 * @throws \OCP\PreConditionNotMetException
 	 */
 	protected function execute(InputInterface $input, OutputInterface $output) {
-		$uid = $input->getArgument('uid');
-
-		/** @var $user \OCP\IUser */
-		$user = $this->userManager->get($uid);
-
-		if ($user === null) {
-			$output->writeln("<error>Unknown user: $uid</error>");
-			return self::EX_NOUSER;
-		}
-
-		if (!$user->canChangePassword()) {
-			$output->writeln("<error>The user's backend doesn't support password changes. The password cannot be expired for user: $uid</error>");
-			return self::EX_GENERAL_ERROR;
+		$groups = $input->getOption('groups');
+		$allUsers = $input->getOption('all');
+
+		if ($allUsers !== false) {
+			$users = $this->userManager->search('');
```

Comment:
can we use `callForAllUsers` instead ? the search will not scale for bigger setups